ddtrace/tracer.py

```diff
@@ -174,24 +202,34 @@ def start_span(self, name, child_of=None, service=None, resource=None, span_type
                 span_type=span_type,
             )
 
-            span.set_tag(system.PID, getpid())
-
-            #    http://pypi.datadoghq.com/trace/docs/#distributed-tracing
-            parent_trace_id, parent_span_id = context._get_parent_span_ids()
-            if parent_trace_id:
-                span.trace_id = parent_trace_id
-
-            if parent_span_id:
-                span.parent_id = parent_span_id
-
-            self.sampler.sample(span)
+            span.sampled = self.sampler.sample(span)
+            if span.sampled:
+                # When doing client sampling in the client, keep the sample rate so that we can
+                # scale up statistics in the next steps of the pipeline.
+                if isinstance(self.sampler, RateSampler):
+                    span.set_metric(SAMPLE_RATE_METRIC_KEY, self.sampler.sample_rate)
+
+                if self.priority_sampler:
+                    if self.priority_sampler.sample(span):
+                        span.set_sampling_priority(1)
+                    else:
+                        span.set_sampling_priority(0)
+            else:
+                if self.priority_sampler:
+                    # If dropped by the local sampler, distributed instrumentation can drop it too.
+                    span.set_sampling_priority(0)
 
         # add common tags
         if self.tags:
             span.set_tags(self.tags)
+        if not span._parent:
+            span.set_tag(system.PID, getpid())
+
+        # TODO: add protection if the service is missing?
```

Comment:
yes but it's another topic that would be handled by our compatibility layer. You can safely remove this `TODO` since I already track this.